### PR TITLE
#161382350 Users should be able to presented with a reset password link

### PR DESCRIPTION
**Pull request by emukungu**
#### What does this PR do?
Provide a user interface to enable the user reset his/her password

#### Description of Task to be completed?
Users should be able to presented with a reset password link

#### How should this be manually tested?
 - Clone the repository.
 - Install application dependencies using`yarn`
 - start the server using `yarn start:dev`
 - Enter `localhost:8080/forgot_password` in the browser
 - Enter the email and submit
 - Check the inbox of the email address provided
 - Click on the link in the inbox
 - You will be redirected to another page to provide the new password
 - Enter the new password and submit(the link can only be used once)

#### Any background context you want to provide?
The link in the backend should redirect the user to the user interface in the frontend

#### What are the relevant pivotal tracker stories?
#161382350

#### Screenshots (if appropriate)

Page to allow user provide email to which the reset link is sent
nefg

Page to allow user put their new password
newcp

### CODE REVIEW

**Comment by shakirandagire** [comment]
Could you squash your commit messages

File: src/actions/resetPasswordAction.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+});
+
+export const userEmail = res => ({
+  type: "USER EMAIL INPUT",
+  payload: res
+});
+
+export const sendEmailLink = data => dispatch => {
+  /*Sends a pasword reset link*/
+  fetch("http://localhost:8000/api/users/password_reset/", {
```
Comment (shakirandagire) [inline src/actions/resetPasswordAction.jsx]: Could you put the url in a dotenv file

File: src/actions/resetPasswordAction.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    method: "POST",
+    mode: "cors",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(data)
+  })
+    .then(response => {
+      return response.json();
+    })
+    .then(value => {
+      // swal.showLoading();
```
Comment (shakirandagire) [inline src/actions/resetPasswordAction.jsx]: Remove commented out code

**Review by shakirandagire — COMMENTED** [review]
Could you put the snapshots in a gitignore file.

**Comment by shakirandagire** [comment]
To increase coverage could you right some end to end tests for your task

File: src/actions/resetPasswordAction.jsx (hunk 2 above)
Comment (emukungu) [reply src/actions/resetPasswordAction.jsx]: removed commented code

File: src/actions/resetPasswordAction.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import swal from "sweetalert2";
+
+export const sendEmailLink = data => dispatch => {
+  /*Sends a pasword reset link*/
+  return fetch("http://localhost:8000/api/users/password_reset/", {
```
Comment (bozicschucky) [inline src/actions/resetPasswordAction.jsx]: Can you kindly remove all the localhost urls from your actions .

Comment (emukungu) [reply src/actions/resetPasswordAction.jsx]: @chucky, fixing all reviews

**Review by johnkegz — APPROVED** [review]

File: src/actions/resetPasswordAction.jsx (hunk 1 above)
Comment (emukungu) [reply src/actions/resetPasswordAction.jsx]: @shakirandagire , removed localhost url

Comment (emukungu) [reply src/actions/resetPasswordAction.jsx]: @shakirandagire , if I put the url in env, the mock test won't be able to pick up the fetch function

File: src/index.html [hunk 4]
```diff
@@ -3,10 +3,27 @@
...
+	<!-- Material Design Bootstrap -->
+	<link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.5.15/css/mdb.min.css" rel="stylesheet">
+	<link href="https://cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/toastr.min.css" rel="stylesheet">
+
   <title>Authors' Haven</title>
-</head>
 <body>
   <div id="root"></div>
 </body>
+    <!-- JQuery -->
```
Comment (joshNic) [inline src/index.html]: Please move all the script tags to the end of the body tag or the head tag right now they are outside of both the body and head tags

File: tests/integerationTests/ResetComponents.test.jsx~HEAD [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
```
Comment (joshNic) [inline tests/integerationTests/ResetComponents.test.jsx~HEAD]: could you please remove this file

File: tests/unitTests/ResetActions.test.jsx~HEAD [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import React from "react";
```
Comment (joshNic) [inline tests/unitTests/ResetActions.test.jsx~HEAD]: duplicate file could you please remove it

**Review by joshNic — CHANGES_REQUESTED** [review]
@emukungu PR looks good but could you please address the issues I have pointed out for it to be merged.

File: tests/unitTests/ResetActions.test.jsx~HEAD (hunk 6 above)
Comment (emukungu) [reply tests/unitTests/ResetActions.test.jsx~HEAD]: resolved

File: tests/integerationTests/ResetComponents.test.jsx~HEAD (hunk 5 above)
Comment (emukungu) [reply tests/integerationTests/ResetComponents.test.jsx~HEAD]: resolved

File: src/index.html (hunk 4 above)
Comment (emukungu) [reply src/index.html]: resolved

**Review by joshNic — APPROVED** [review]

**Pull request merged**